source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -66,11 +66,21 @@ void Filter::initiateCall(const Http::RequestHeaderMap& headers) {
     break;
   }
 
+  double hits_addend = 0;
+  const auto& request_metadata = callbacks_->streamInfo().dynamicMetadata().filter_metadata();
+  const auto filter_it = request_metadata.find("envoy.ratelimit");
+  if (filter_it != request_metadata.end()) {
+    const auto field_it = filter_it->second.fields().find("hits_addend");
+    if (field_it != filter_it->second.fields().end()) {
+      hits_addend = field_it->second.number_value();
```

Comment:
There is no any guarantee that the `hits_addend` is number type?

Reply:
The default returned value from this function is `0`. So in the case that it's any other type `hits_addend` will simply remain `0`.

Reply:
if we expect the `hits_addend` is number but it acutally is other types, then I think this is a bug and at least we should add an `ENVOY_BUG` here.

---

Basically, we expect specific typed data here. This is why the `FilterState` is recommend. We also provide a `set_filter_state` filter to modify the filter state.

And if you prefer the dynamic metadata and insist on that point, I won't stop if matt approve to that.